src/test/java/com/virtuslab/archunit/BackgroundTaskEnqueuingTestSuite.java

```diff
@@ -0,0 +1,123 @@
+package com.virtuslab.archunit;
+
+import static com.tngtech.archunit.lang.syntax.ArchRuleDefinition.methods;
+
+import java.util.Arrays;
+
+import com.intellij.openapi.progress.Task;
+import com.tngtech.archunit.core.domain.AccessTarget;
+import com.tngtech.archunit.core.domain.JavaClass;
+import com.tngtech.archunit.core.domain.JavaMethod;
+import com.tngtech.archunit.lang.ArchCondition;
+import com.tngtech.archunit.lang.ConditionEvents;
+import com.tngtech.archunit.lang.SimpleConditionEvent;
+import io.vavr.collection.List;
+import lombok.experimental.ExtensionMethod;
+import lombok.val;
+import org.junit.Test;
+
+import com.virtuslab.qual.async.ContinuesInBackground;
+import com.virtuslab.qual.async.DoesNotContinueInBackground;
+import com.virtuslab.qual.guieffect.IgnoreUIThreadUnsafeCalls;
+
+@ExtensionMethod(Arrays.class)
+public class BackgroundTaskEnqueuingTestSuite extends BaseArchUnitTestSuite {
+
+  private static final String ContinuesInBackgroundName = "@" + ContinuesInBackground.class.getSimpleName();
+
+  private static List<String> extractWhitelistedMethodsFromAnnotation(JavaMethod method) {
+    if (method.isAnnotatedWith(IgnoreUIThreadUnsafeCalls.class)) {
+      return List.of(method.getAnnotationOfType(IgnoreUIThreadUnsafeCalls.class).value());
+    } else {
+      return List.empty();
+    }
+  }
+
+  @Test
+  public void only_continues_in_background_methods_should_call_other_continues_in_background_methods() {
```

Comment:
Mostly shamelessly copied from `UIThreadUnsafeMethodInvocationsTestSuite`... I'm sticking to _twice repeated? copy! thrice repeated? extract_ rule that I've learned in the olden days at Google ;p